### ISSUE TITLE
Probably bugs: Issues moving and expanding notes inside a notebook

### ISSUE THREAD

**Issue by anotherboringuser** [orgzly/orgzly-android#553]
Hi there, 
Today I thought it would be a good idea to restructure my ToDo-List in Orgzly which ended in a mess. I moved a lot of notes inside a notebook to make them sub-notes of other notes, but I experienced some issues or at least behavior I don’t understand: 
1.	When I selected a note and then moved it up or down in a notebook, I saw that another note started to “stick” to the note I moved and moved with it. I was unable to separate the two. Only one was selected and the other one was not a sub-note (at least it did not look like that). 
2.	After expanding a note that has some sub-notes, it “swallowed” notes that where below it and resulted in them getting stuck as sub-notes that I could show and hide with the + or – icons on the right of a note. Some of the sub-notes where on the same level vertically as the note that they are a sub note of. I was unable to get them out of there again. There were also notes that suddenly had the + icon on the right but that did not show anything when I tapped on the +. They looked like they had opened their sub-notes but there were no sub-notes and I never put any there. 
When tapping on the + of the green note, all blue notes appear. The light blue one looks like it has sub-notes but it doesn’t. 
![Screenshot_2019-06-09-17-44-29_1_1](https://user-images.githubusercontent.com/44035735/59161774-15d6fc00-8ae7-11e9-9b27-d5f5309fca24.png)

3.	I have some private information in the notebook where I experienced the issues, so I can’t share it with you (in case that would make troubleshooting easier). That’s why I tried reproducing the problem in the “getting started with Orgzly” note, but it didn’t work. I noticed however that I was able to create a note (it says "suchen") that could not be moved at all anymore (tapping the arrows results in no action). 
![Screenshot_2019-06-09-17-29-34](https://user-images.githubusercontent.com/44035735/59161781-27b89f00-8ae7-11e9-8759-f890b0e4e35b.png)
I use Orgzly every day to outsource things from my brain, but now I don’t really trust it anymore. And I don’t want to touch anything because it could make things worse. I would be very happy if someone told me how to solve the issues or prevent them. Or did I just use the app incorrectly? 

You can download my "Getting started with Orgzly" notebook here (I was unable to upload the .org file to GitHub): https://send.firefox.com/download/2f6342ae5b314b1b/#7_Uv3vi0e89Z7Y2Y8iH-kw 

My issues might be similar to #524 https://github.com/orgzly/orgzly-android/issues/524 
Note: 
I am not a native speaker so I don’t know if the terms I used are understandable. Just ask if you don’t know what I’m talking about. 
I’m using Orgzly Version 1.7.4 on Android 8.0.0 and I have automatic syncing enabled (when notes are created or the app is started or continued).

**Comment by nevenz** [comment]
Thanks for the detailed report. Looking into this.

BTW, did you use Orgzly before v1.7 (released at the beginning of March) ?

**Comment by anotherboringuser** [comment]
Yes, I've used Orgzly since the end of last year, but I never had big issues before.

**Comment by nevenz** [comment]
> Yes, I've used Orgzly since the end of last year, but I never had big issues before.

I'm trying to determine if database migration done for v1.7 messed up some structure-related values, or it's a current issue with moving/cutting/pasting/refiling.

If you can start with clean notebooks (by deleting them or clearing the local database and re-syncing or re-importing them), it might give us a clue if you're still seeing (or not) the issue.

**Comment by nevenz** [comment]
Oh and did you use cut/copy/paste? Or only moving (arrow icons)?

**Comment by anotherboringuser** [comment]
I copied the .org file of the problematic notebook to another phone with Orgzly freshly installed, and there some of the issues disappeared: 
2. 1. I wasn’t able to unfold the notes that got “swallowed” by the green note with the + icon next to it anymore, they appear as individual notes below like I expected them to. 
2. 2. The notes that look like they could be unfolded (I called that "expanded" in my original post) but that did not have any sub-notes in them (like the light blue note) are normal sub-notes now that don’t look like they could be unfolded. 
I was able to organize everything and put the notes where they belong on the freshly installed Orgzly phone, so there it looks like nothing ever happened. 

When I loaded the “Getting started with Orgzly” note on the phone with freshly installed Orgzly, I could see the problem (3.), but I could move the note to a normal position with one tap in the <-- arrow, which is not possible in the original note on the phone with the problems. 

As far as I remember I only used refile (from the note with problems to other notebooks) and moved notes with the arrow icons. I copied text inside notes with the normal android copy features, but I don’t remember having used the cut / copy / paste icons that appear at the top after long pressing a note. 

Shortly before I experienced issues yesterday I think I changed my sync settings from “note created” and “note updated or deleted” to “note created” and “app started or resumed” because I Orgzly synced the notes every time I tapped on a moving arrow, which I did a lot while moving notes up and down the notebook. I don’t know if this is related to my issues though.
